src/flask/routes/security_route_middleware.py

```diff
@@ -0,0 +1,54 @@
+from flask import Blueprint, request, make_response
+
+import jwt
+
+from src.constants.http_status_code import HttpStatus
+from src.env_config import get_env_config_instance
+from src.exceptions.exceptions import UnAuthorizedEndpoint
+from src.lib.repositories.impl_v2.oauth2_repository_impl import Oauth2RepositoryImpl
+from src.utils.oauth2_util import validate_roles_and_scopes
+
+
+def setup_security_route_middleware(ioc):
+
+    security_route_middleware_blueprint = Blueprint(
+        "security_route_middleware_blueprint", __name__
+    )
+    oauth2_repository = ioc.get(Oauth2RepositoryImpl)
+
+    @security_route_middleware_blueprint.before_app_request
+    def before_request_middleware():
+        endpoint_request = request
+        env_config = get_env_config_instance()
+
+        try:
+
+            protected = validate_roles_and_scopes(
```

Comment:
This is wrong, first you should check if the endpoint is protected and then perform validations. Make these oprations explicit, don't do validation of role and scopes along with checking if the endpoint is protected.

Reply:
done.